adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/ADLSBatchSink.java

```diff
@@ -140,6 +166,12 @@ protected void validate() {
       if (!containsMacro("path") && !path.startsWith("adl://")) {
         throw new IllegalArgumentException("Path must start with adl:// for ADLS input files.");
       }
+      if (TEXT.equals(outputFormat) && textDelimiter == null) {
+        throw new IllegalArgumentException("Field delimiter must be specified for text format output files.");
+      }
+      if (AVRO.equals(outputFormat) && schema == null) {
```

Comment:
need to check that output format and schema are not macros before doing this check. Otherwise, at configure time, they will be null and this will incorrectly error.